Data/Map/Internal.hs

```diff
@@ -2661,7 +2692,11 @@ mergeA
 -- @only2@ are 'id' and @'const' 'empty'@, but for example @'map' f@,
 -- @'filterWithKey' f@, or @'mapMaybeWithKey' f@ could be used for any @f@.
 
+#if __GLASGOW_HASKELL__ >= 800
+mergeWithKey :: (HasCallStack, Ord k)
```

Comment:
Ehhhhh... Not sure we should bother with this one. `mergeWithKey` is for users who don't mind the risk of being shot in the foot.